### DISCUSSION TITLE
Using this package as a dev dependency

### DISCUSSION THREAD

**Discussion by mrk-j** [themsaid/laravel-mail-preview#71] [Ideas]
Hi,

We were using @themsaid's package as a dev dependency. Out of the box this is not possible with this package because of the middleware, in production the package is not installed and triggers a `Target class [Spatie\MailPreview\Http\Middleware\AddMailPreviewOverlayToResponse] does not exist.` Exception.

I have two solutions for this problem:

1. Move the middleware from the package to the project and check if the middleware is not running in `production`.
2. Conditionally add the middleware to the web group depending on the environment like this:

```php
<?php

namespace App\Providers;

use App\Http\Kernel;
use Illuminate\Support\ServiceProvider;

class AppServiceProvider extends ServiceProvider
{
    /**
     * Bootstrap any application services.
     *
     * @return void
     */
    public function boot(Kernel $kernel)
    {
        if ($this->app->environment('local')) {
            $kernel->appendMiddlewareToGroup('web', \Spatie\MailPreview\Http\Middleware\AddMailPreviewOverlayToResponse::class);
        }
    }
}
```

For now we went with number two.

What is your opinion on this?

**Comment by BrendanTWhite** [comment]
I gave up and just made it a regular dependency, not a dev dependency.
